cds/modules/deposit/api.py

```diff
@@ -855,6 +855,10 @@ def _publish_edited(self):
         """Rename subtitles and publish."""
         self['_files'] = self.files.dumps()
         self._rename_subtitles()
+
+        # Mint the doi if necessary
+        doi_minter(record_uuid=self.id, data=self)
```

Comment:
Just move this up in the `CDSDeposit`.

Then, to improve the code, just add an extra if to call the celery task only if needed:
```
if record.get('doi'):
    datacite_register.delay(recid_pid.pid_value, str(record.id))
```
in `cds.modules.deposit.receivers`